Server/Services/IStackwalkSymbolPreparer.cs

```diff
@@ -22,7 +23,7 @@ public interface IStackwalkSymbolPreparer
 
 public class StackwalkSymbolPreparer : IStackwalkSymbolPreparer
 {
-    private static readonly SemaphoreSlim GlobalSymbolPrepareLock = new(1);
+    private static readonly AsyncNonKeyedLocker GlobalSymbolPrepareLock = new(1);
```

Comment:
What's the reasoning for replacing SemaphoreSlim with this custom lock type here? (I also have the same question about the discord bot service getting the same treatment).

Reply:
It uses the same 'interface' as the AsyncKeyedLock from the same library (i.e. not adding a new dependency), making your code more streamlined, easier to read and avoids certain issues like forgetting to released which is what happened in BuildWebSocketHandler.

Reply:
I see, so the benefit is that the API is easier to use and to not cause bugs accidentally? That is certainly an improvement. The downside here being that instead of using a pure C# standard library feature, this now depends on an external library. I've been burned in the past by having to ditch libraries that are no longer maintained. I'm not saying that this is a huge risk, you've been taking good care of the async lock library development, but still something to consider for me. 
I'll move forward with the idea that this change is fine and check the changes in detail.

Reply:
I understand your concerns but I personally use my own library in so many projects that the only chance it will ever stop being maintained is if Microsoft make something similar as part of their framework, and in which case I'd create a new target framework so that it would internally use that for that target framework and above.